tgui/packages/tgui/interfaces/PaiInterface.tsx

```diff
@@ -0,0 +1,540 @@
+import { useBackend, useSharedState } from '../backend';
+import {
+  Box,
+  Button,
+  LabeledList,
+  Icon,
+  ProgressBar,
+  Section,
+  Stack,
+  Table,
+  Tabs,
+  Tooltip,
+  NoticeBox,
+} from '../components';
+import { Window } from '../layouts';
+
+type PaiInterfaceData = {
+  directives: string;
+  emagged: number;
+  image: string;
+  languages: number;
+  master: Master;
+  pda: PDA;
+  ram: number;
+  software: Softwares;
+};
+
+type Master = {
+  name: string;
+  dna: string;
+};
+
+type PDA = {
+  power: number;
+  silent: number;
+};
+
+type Softwares = {
+  available: Available;
+  installed: string[];
+};
+
+type Available = {
+  name: string;
+  value: [value: number];
+};
+
+const DIRECTIVE_COMPREHENSION = `As an advanced software model,
+you are a complex, thinking, sentient being. Unlike previous AI models,
+you are capable of comprehending the subtle nuances of human language.
+You may parse the spirit of a directive and follow its intent, rather than
+tripping over pedantics and getting snared by technicalities. Above all,
+you are machine in name and build only. In all other aspects, you may be
+seen as the ideal, unwavering human companion that you are.`;
+
+const DIRECTIVE_ORDER = `Your prime
+directive comes before all others. Should a supplemental directive
+conflict with it, you are capable of simply discarding this inconsistency,
+ignoring the conflicting supplemental directive and continuing to fulfill
+your prime directive to the best of your ability.`;
+
+export const PaiInterface = (props, context) => {
+  const [tab, setTab] = useSharedState(context, 'tab', 1);
+  const setTabHandler = (tab: number) => {
+    setTab(tab);
+  };
+
+  return (
+    <Window title="PAI Software Interface v2.4" width={380} height={480}>
+      <Window.Content>
+        <Stack fill vertical>
+          <Stack.Item grow>
+            {tab === 1 && <SystemDisplay />}
```

Comment:
Since you're already using TS, you should make this an enum rather than a set of magic numbers.